### PR TITLE
crate_universe: Only compile label regex once.

### PR DESCRIPTION
**Pull request by criemen**
I hit this when profiling for something else, we spent 1.1sec of 1.3sec of `cargo-bazel generate` compiling this regular expression. It seems to be quite bad on the regex compiler, as even compiling it once takes 8ms.

I had to add the `once_cell` crate to unwrap the Result type - the corresponding stdlib function is still nightly-only unfortunately. However, that crate was already in the transitive dependency tree, so it's a net-same change.

### CODE REVIEW

**Comment by criemen** [comment]
Can someone please push the correct, re-vendored dependencies here? I spent now 30min trying to get this to work locally, but it's not working for me.

**Comment by criemen** [comment]
Revendoring the dependencies helped with some CI jobs, but the BCR jobs still complain about a lockfile being out of date. I ran the suggested command locally in `examples/bzlmod/hello_world`, but that doesn't seem to satisfy CI. Can someone help me out here?

File: crate_universe/Cargo.toml [hunk 1]
```diff
@@ -49,6 +49,7 @@ textwrap = "0.16.0"
 toml = "0.8.10"
 tracing = "0.1.40"
 tracing-subscriber = "0.3.17"
+once_cell = "1.19.0"
```
Comment (UebelAndre) [inline crate_universe/Cargo.toml:52]: nit, can you sort this into the list above?

**Review by UebelAndre — CHANGES_REQUESTED** [review]
Thank you for this! I think there are some extra things you'll need manually to update on the bzlmod side of this. See https://github.com/bazelbuild/rules_rust/issues/2368

**Comment by criemen** [comment]
Thanks! CI finally passed on this.
That was harder than expected given the size of the core change...

**Review by UebelAndre — APPROVED** [review]
Thank you! For this change I probably would have just regenerated dependencies for `crate_universe` directly but thank you for going above and beyond here 😄

**Pull request merged**